src/tmate.py

```diff
@@ -56,23 +63,53 @@ class FingerprintError(Exception):
     """Represents an error with generating fingerprints from public keys."""
 
 
-def install_dependencies() -> None:
+def _setup_docker(proxy_config: typing.Optional[state.ProxyConfig] = None) -> None:
+    """Install and configure proxy settings for docker if available.
+
+    Args:
+        proxy_config: The proxy configuration to enable for dockerd.
+
+    Raises:
+        DependencySetupError: if there was a problem installing/setting up Docker.
+    """
+    if proxy_config:
+        environment = jinja2.Environment(
+            loader=jinja2.FileSystemLoader("templates"), autoescape=True
+        )
+        docker_template = environment.get_template("docker_daemon.json.j2")
+        daemon_config = docker_template.render(
+            HTTP_PROXY=proxy_config.http_proxy,
+            HTTPS_PROXY=proxy_config.https_proxy,
+            NO_PROXY=proxy_config.no_proxy,
+        )
+        DOCKER_DAEMON_CONFIG_PATH.parent.mkdir(parents=True, exist_ok=True)
+        DOCKER_DAEMON_CONFIG_PATH.touch(exist_ok=True)
+        DOCKER_DAEMON_CONFIG_PATH.write_text(daemon_config, encoding="utf-8")
+
+    try:
+        apt.add_package("docker.io", update_cache=True)
+    except (apt.PackageNotFoundError, apt.PackageError, subprocess.CalledProcessError) as exc:
```

Comment:
I don't see in the lib `add_package`raises `CalledProcessError`. Am I missing something here?